argoverse/evaluation/eval_detection.py

```diff
@@ -0,0 +1,329 @@
+
+                metrics[dt_tp_indices, n_threshs : n_threshs + NUM_TP_METRICS] = np.vstack(
+                    (trans_error, scale_error, orient_error)
+                ).T
+        return metrics, scores
+
+    def summarize(
+        self, data: DefaultDict[str, np.ndarray], cls_to_ninst: DefaultDict[str, int]
+    ) -> DefaultDict[str, List]:
+        """Calculate and print the detection metrics.
+
+        Args:
+            data: The aggregated data used for summarization.
+            cls_to_ninst: Map of classes to number of instances.
+
+        Returns:
+            summary: The summary statistics.
+        """
+        summary: DefaultDict[str, List] = defaultdict(list)
+        recalls_interp = np.linspace(0, 1, self.detection_cfg.n_rec_samples)
+        num_ths = len(self.detection_cfg.affinity_threshs)
+        if not self.figures_fpath.is_dir():
+            self.figures_fpath.mkdir(parents=True, exist_ok=True)
+
+        for cls_name, cls_stats in data.items():
+            ninst = cls_to_ninst[cls_name]
+            ranks = cls_stats[:, -1].argsort()[::-1]
+            cls_stats = cls_stats[ranks]
+
+            for i, _ in enumerate(self.detection_cfg.affinity_threshs):
+                tp = cls_stats[:, i].astype(bool)
+
+                cumulative_tp = np.cumsum(tp, dtype=np.int)
+                cumulative_fp = np.cumsum(~tp, dtype=np.int)
+                cumulative_fn = ninst - cumulative_tp
+
+                precisions = cumulative_tp / (cumulative_tp + cumulative_fp + np.finfo(float).eps)
+                recalls = cumulative_tp / (cumulative_tp + cumulative_fn)
+
+                precisions = interp(precisions)
+                precisions_interp = np.interp(recalls_interp, recalls, precisions, right=0)
+
+                ap_th = precisions_interp.mean()
+                summary[cls_name] += [ap_th]
+
+                if self.detection_cfg.save_figs:
+                    self.plot(recalls_interp, precisions_interp, cls_name)
+
+            # AP Metric
+            ap = np.array(summary[cls_name][:num_ths]).mean()
+
+            # Select only the true positives for each instance.
+            tp_metrics_mask = ~np.isnan(cls_stats[:, num_ths : num_ths + NUM_TP_METRICS]).all(axis=1)
+
+            # If there are no true positives set tp errors to their maximum values due to normalization below)
+            if ~tp_metrics_mask.any():
+                tp_metrics = self.detection_cfg.tp_normalization_terms
+            else:
+                # Calculate TP metrics.
+                tp_metrics = np.mean(cls_stats[:, num_ths : num_ths + NUM_TP_METRICS][tp_metrics_mask], axis=0)
+
+            # Convert errors to scores
+            tp_scores = 1 - (tp_metrics / self.detection_cfg.tp_normalization_terms)
+
+            # Compute Composite Detection Score (CDS)
+            cds = ap * tp_scores.mean()
+
+            summary[cls_name] = [ap, *tp_metrics, cds]
+
+        logger.info(f"summary = {summary}")
```

Comment:
we need to save a json file with the results for the leaderboard, right @janesjanes ? are we doing that now?

Reply:
bump @janesjanes

Reply:
@janesjanes is it sufficient for your needs to just do `.json()` from the pandas dataframe?

Reply:
Patsorn confirmed she will use a wrapper around the evaluator, so json is not necessary